nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -861,6 +979,95 @@ impl BundleCollection {
         }
         return Ok(());
     }
+
+    async fn collect_sp_ereports(
+        &self,
+        filters: &EreportFilters,
+        dir: &Utf8Path,
+        ereports_written: &mut usize,
+    ) -> anyhow::Result<()> {
+        let mut paginator = Paginator::new(
+            datastore::SQL_BATCH_SIZE,
+            dropshot::PaginationOrder::Ascending,
+        );
+        while let Some(p) = paginator.next() {
```

Comment:
I wonder if we should have an API to collect these more gradually - we *could*, for some smallish N and large M, grab "N of M" ereports, write them to the bundle, and then grab the next batch.

For now, it seems like we're loading all ereports requested into memory, which could be really large. Maybe this is fine to start? But it's probably worth filing an issue, if we don't want to optimize this now.

Reply:
The query is paginated, but I just used the default `SQL_BATCH_SIZE`, which is pretty big. We could just use a smaller value here to limit the amount that's held in memory?

Reply:
ahhh nevermind, I think I read this incorrectly - using SQL_BATCH_SIZE seems totally fine to me, that's a constant bound on how much we're loading.